Prometheus ci startup timeout error

Transcendental-Programmer opened the pull request:
## Description

This PR includes the following changes:

1. Ensures the CI workflow fails when any test fails, improving reliability.
2. Sets the correct Git email for automated commits to avoid identity issues in logs.
3. Fixes Prometheus installation by setting the required `maximumStartupDurationSeconds` parameter, preventing startup errors during deployment.

## Documentation

* [ ] **Is documentation needed for this update?**
  (If checked, a documentation PR must be created and merged in the [[website repository](https://github.com/krkn-chaos/website/)](https://github.com/krkn-chaos/website/).)

## Related Documentation PR (if applicable)

<!-- Add the link to the corresponding documentation PR in the website repository -->  


---

Transcendental-Programmer commented:
@tsebastiani does this fix the issue in https://github.com/krkn-chaos/krkn/issues/827 and i also want to solve the same issue in other repos if you could review this pr i will proceed for the other repos

paigerube14 commented:
I approved your functional test run but I'm not sure why it's not pulling up in the checks list but it looks like you can only set one "run" in your workflow 

https://github.com/krkn-chaos/krkn/actions/runs/15304867377

Transcendental-Programmer commented:
> I approved your functional test run but I'm not sure why it's not pulling up in the checks list but it looks like you can only set one "run" in your workflow
> 
> https://github.com/krkn-chaos/krkn/actions/runs/15304867377

yes i fixed that and also squashed my commits to a single one.

Transcendental-Programmer commented:
> @Transcendental-Programmer thanks for the updates on this PR but something still doesn't look quite right. Can you take another look here. TIA
> 
> ```
> namespace/prometheus-k8s created
> Error: INSTALLATION FAILED: 1 error occurred:
> 	* Prometheus.monitoring.coreos.com "kind-prometheus-kube-prome-prometheus" is invalid: spec.maximumStartupDurationSeconds: Invalid value: 0: spec.maximumStartupDurationSeconds in body should be greater than or equal to 60
> ```

hi @paigerube14 that's the exact error i am trying to fix through the pr, since the current ci file which does not have my changes will be checking the correctness of the ci tests so it is intended to fail.

paigerube14 commented:
@Transcendental-Programmer if you look at [this run](https://github.com/krkn-chaos/krkn/actions/runs/15305953427/job/43059278807#step:5:1) I think it is using your changes but still giving error

Transcendental-Programmer commented:
Hi @paigerube14 , thanks for pointing that out. I took another look, and it turns out the error was due to a super tiny issue : a missing backslash :) , which is why the command 

```
--set prometheus-node-exporter.service.type=NodePort (\) --> missing backslash 
          --set prometheus.prometheusSpec.maximumStartupDurationSeconds=300
```

was being treated as a separate command rather than being part of the helm install.
